src/operator/tensor/matrix_op-inl.h

```diff
@@ -1217,6 +1218,83 @@ void Slice(const nnvm::NodeAttrs& attrs,
   });
 }
 
+// slice the indptr of a csr
+struct SliceCsrIndPtr {
+  template<typename IType>
+  MSHADOW_XINLINE static void Map(int i, IType* out, const IType* in, IType* base) {
```

Comment:
I see your point. It's better to put a const before IType* since you will not change base.